### PR TITLE
Updated anchor-pdas to be in sync with latest code changes

### PR DESCRIPTION
**Pull request by ASCorreia**
### Problem
anchor-pdas.md was not in sync with latest code changes

### Summary of Changes
Added the needed changes to the account struct, contexts and tests in order to make the markdown file aligned with the latest code changes

### CODE REVIEW

File: content/courses/onchain-development/anchor-pdas.md [hunk 1]
```diff
@@ -454,7 +454,49 @@ pub struct MovieAccountState {
...
+    We add the STRING_LENGTH_PREFIX twice to the space to account for the title and description string prefix.
+    We need to add the length of the title and description to the space upon initialization.
+ */
+impl Space for MovieAccountState {
+    const INIT_SPACE: usize = ANCHOR_DISCRIMINATOR + PUBKEY_SIZE + U8_SIZE + STRING_LENGTH_PREFIX + STRING_LENGTH_PREFIX;
+}
+```
+
+This allows us to declare the space required by this account by defining the
+INIT_SPACE constant (required by the Space trait)
```
Comment (nickfrosty) [inline content/courses/onchain-development/anchor-pdas.md:471]:
```suggestion
`INIT_SPACE` constant (required by the `Space` trait).
```

File: content/courses/onchain-development/anchor-pdas.md [hunk 2]
```diff
@@ -454,7 +454,49 @@ pub struct MovieAccountState {
 }
 ```
 
-#### 3. Add Movie Review
+For this account struct, we will be implementing the space trait:
```
Comment (nickfrosty) [inline content/courses/onchain-development/anchor-pdas.md:457]: can you add some details of what the space trait is for and why its good to use. the code block comment covers it, but I think having it in the regular paragraph will be more usable by readers

File: content/courses/onchain-development/anchor-pdas.md [hunk 3]
```diff
@@ -454,7 +454,49 @@ pub struct MovieAccountState {
...
+    #[msg("Movie Description too long")]
+    DescriptionTooLong,
+}
+```
+
+The `#[error_code]` macro will generate error types to be used as return types
+from our instruction handlers.
+
+Don't worry too much about custom errors for now, as they will be covered with
+more detail in the next chapter
```
Comment (nickfrosty) [inline content/courses/onchain-development/anchor-pdas.md:497]:
```suggestion
more detail in the next chapter.
```

File: content/courses/onchain-development/anchor-pdas.md [hunk 4]
```diff
@@ -471,6 +513,15 @@ Within the instruction logic, we’ll populate the data of the new `movie_review
...
 `initializer` account from the instruction context.
 
+We will also perform some checks, using the `require!` macro, to make sure that:
+
+- The rating is between 1 and 5
+- The title is no longer than 20 characters
+- The description is no longer than 50 characters
+
+The `require!` macro will perform a check and throw a custom error in case that
+check is not successful
```
Comment (nickfrosty) [inline content/courses/onchain-development/anchor-pdas.md:523]:
```suggestion
check is not successful.
```

File: content/courses/onchain-development/anchor-pdas.md [hunk 5]
```diff
@@ -512,7 +572,9 @@ Remember, you'll need the following macros:
 
 The `movie_review` account is a PDA that needs to be initialized, so we'll add
 the `seeds` and `bump` constraints as well as the `init` constraint with its
-required `payer` and `space` constraints.
+required `payer` and `space` constraints. Regarding the required space, we will
+be using the `INIT_SPACE` constant that we defined in the account struct, and we
+will add the String length of the both the title and the description
```
Comment (nickfrosty) [inline content/courses/onchain-development/anchor-pdas.md:577]:
```suggestion
will add the string length of the both the title and the description.
```

**Review by nickfrosty — COMMENTED** [review]
Thanks for updating this. Some small nits on the updates. Mostly using single backticks, fixing captialization, and adding the missing periods at the end of sentences

**Comment by nickfrosty** [comment]
@ASCorreia you could also update this to use the new `<Steps>` component (it just got added a few days ago)

You can see it in action on this page now: https://solana.com/docs/intro/quick-start#create-playground-wallet

If you want to add it, simply wrap the contents of the "Lab" section with `<Steps>` and `</Steps>`. Then update each of the headings within to use h3 tags. Then it works!

Example:

```
## Lab

Let’s practice the concepts we’ve gone over in this lesson by creating a Movie
Review program using the Anchor framework.

This program will allow users to:

- Use a PDA to initialize a new movie review account to store the review
- Update the content of an existing movie review account
- Close an existing movie review account

<Steps>

### Create a new Anchor project

...other content

### MovieAccountState

... the rest of the content
</Steps>
```

Note: If you do this, make sure to remove the number counter from the heading content since the step counter number is generated automatically. Also remove any single backticks from the headings

File: content/courses/onchain-development/anchor-pdas.md (hunk 2 above)
Comment (ASCorreia) [reply content/courses/onchain-development/anchor-pdas.md:459]: Agreed. Added more details regarding the space trait

**Comment by ASCorreia** [comment]
@nickfrosty Reworked according to review :)

**Review by nickfrosty — APPROVED** [review]
Thanks for the updates @ASCorreia, the step component will be nice!

**Pull request merged**